src/main/fc/core.c

```diff
@@ -524,28 +525,26 @@ void tryArm(void)
             }
 #endif
 
-            if (isModeActivationConditionPresent(BOXFLIPOVERAFTERCRASH)) {
-                // Set motor spin direction
-                if (!(IS_RC_MODE_ACTIVE(BOXFLIPOVERAFTERCRASH) || (tryingToArm == ARMING_DELAYED_CRASHFLIP))) {
-                    flipOverAfterCrashActive = false;
-                    if (!featureIsEnabled(FEATURE_3D)) {
-                        dshotCommandWrite(ALL_MOTORS, getMotorCount(), DSHOT_CMD_SPIN_DIRECTION_NORMAL, DSHOT_CMD_TYPE_INLINE);
-                    }
-                } else {
-                    flipOverAfterCrashActive = true;
-#ifdef USE_RUNAWAY_TAKEOFF
-                    runawayTakeoffCheckDisabled = false;
-#endif
-                    if (!featureIsEnabled(FEATURE_3D)) {
-                        dshotCommandWrite(ALL_MOTORS, getMotorCount(), DSHOT_CMD_SPIN_DIRECTION_REVERSED, DSHOT_CMD_TYPE_INLINE);
-                    }
+            // choose crashflip outcome if no arming disabled flags and arm switch enabled... 
+            if (crashFlipModeActive) {
+                if (tryingToArm == ARMING_DELAYED_CRASHFLIP || (!IS_RC_MODE_ACTIVE(BOXCRASHFLIP) && crashFlipSuccessful())) {
+                    // arming delayed, or user switched turtle mode off after successful flip
+                    crashFlipModeActive = false; // continue into normal flight without need to disarm/rearm
+                }
+            } else {
+                if (IS_RC_MODE_ACTIVE(BOXCRASHFLIP)) {
+                    crashFlipModeActive = true; // continue into crashflip mode
                 }
             }
+            const uint8_t spinDirection = crashFlipModeActive ? DSHOT_CMD_SPIN_DIRECTION_REVERSED : DSHOT_CMD_SPIN_DIRECTION_NORMAL;
+            if (!featureIsEnabled(FEATURE_3D)) {
+                dshotCommandWrite(ALL_MOTORS, getMotorCount(), spinDirection, DSHOT_CMD_TYPE_INLINE);
```

Comment:
(old code did not set motor direction when crashflip was not configured. It may cause problems with buggy ESC - just remember it if related issues appear)

Reply:
The old code did repeatedly send a motor direction reversed value every loop, while reversed.  However otherwise it send the 'normal' command once on disarm, and probably repeatedly doing the `ARMING_DELAYED_CRASHFLIP` period.
I just don't know enough about DShot to know if it is best to send these commands just once, or several times, to be sure they get through, or all the time you want a certain direction.
I'll ask around.  The code can be structured to send only once without too much difficulty, but if we were supposed to repeat it a few times to be sure, that would be more complex.
As far as I can tell, the code here won't affect motor direction in 3D mode, and no other mode would reverse the motors.  My testing so far is that it 'seems' OK.

Reply:
im not certain about current code, but in legacy code i believe 3D-on-a-switch crashflip worked differently than feature-3D. i presume all cases need testing.

Reply:
I asked around, and 3D mode works differently.  It definitely does not use the motor direction change commands that are used in crasfhlip mode.  
And I've checked these changes carefully, there is no change that would affect 3D mode; in those respects it is no different from the previous code.

Reply:
sounds good, yes i think it negates `-(normalizedDirection)` or something if i recall, i'm too lazy to look.